ros2cli/ros2cli/command/__init__.py

```diff
@@ -112,3 +125,125 @@ def add_subparsers(
                 command_parser, '{cli_name} {name}'.format_map(locals()))
 
     return subparser
+
+
+class MutableString:
+    """Behave like str with the ability to change the value of an instance."""
+
+    def __init__(self):
+        self.value = ''
+
+    def __getattr__(self, name):
+        return getattr(self.value, name)
+
+    def __iter__(self):
+        return self.value.__iter__()
+
+
+def add_subparsers_on_demand(
```

Comment:
Since `add_subparsers` is deprecated, should there be documentation here?

Reply:
Good point, added in 01616f394b849dbe1e61239da04574efeffc1106.